native/common/jp_exception.cpp

```diff
@@ -563,6 +563,10 @@ JPPyObject PyTrace_FromJavaException(JPJavaFrame& frame, jthrowable th, jthrowab
 		jint lineNum =
 				frame.CallIntMethodA(frame.GetObjectArrayElement(obj, i + 3), context->_java_lang_Integer->m_IntValueID, nullptr);
 
+		// sending -1 will cause issues on Windows
+		if (lineNum<0)
+			lineNum = 0;
+
```

Comment:
@marscher this was a hidden two line fix.   The problem it is addressing is that pytest is trying to do arithmetic.   On linux systems we get -1 they subtract one then add one.  No problem.

On windows system with the same version when we pass -1 to an integer field, we get None in the field.   -2 works, 0 works, but -1 gives None.   Then Pytest fails.

I think that we can fix this back in Java where we pass a line number 0 rather than -1.  But it was easier to patch at this level in case there was some other way that we get invalid line numbers.   

This seems like a typical left hand/right hand.   Making -1 which was supposed to mean "line number not available" as None is completely reasonable.   Trying to check the line numbers in the file (which happens to be 0 counting) for the audit display is perfectly reasonable in pytest.   But both changes can't live together.

Reply:
why the heck is this platform dependent? Oo